### ISSUE TITLE
Updating in a for or while loop

### ISSUE THREAD

**Issue by kushalkolar** [pygfx/rendercanvas#49]
Finally catching up with all this! Taking a look at the examples and wondering if this would be the right way to update the canvas in a for or while loop. It works but I'm not sure if this is the way it's intended to be used, without an event loop running.

```python
import numpy as np
from rendercanvas.auto import RenderCanvas, loop

canvas = RenderCanvas(update_mode="ondemand")
context = canvas.get_context("bitmap")

for i in range(100):
    w, h = canvas.get_logical_size()
    shape = int(h) // 4, int(w) // 4

    bitmap = np.random.uniform(0, 255, shape).astype(np.uint8)
    context.set_bitmap(bitmap)
    canvas.force_draw()
```

related: https://github.com/pygfx/pygfx/discussions/667

I'm starting to think about how to implement this for `fastplotlib`, might help to have a mode where the canvas is updated only when an event occurs or when any buffer in the scene changes? Anyways that will be a separate issue which I'll think about later.

**Comment by almarklein** [comment]
Yes, the way to go is to use `force_draw()`, also see the `demo.py` example:

https://github.com/pygfx/rendercanvas/blob/45843edb32eb332e098bd943e2aaeee052ec8209/examples/demo.py#L44-L52

However, it is meant to be used only in special scenarios because it bypasses the normal event loop and scheduling, which means that there will also be no updates, unless you also flush the events etc. and then you are basically creating your own scheduler ... 

In use-cases where you are running a simulation like https://github.com/pygfx/pygfx/discussions/667, I think it would be better to use something like:
```
async def do_my_simulation():
    for .... 
         do whatever you want to do
         but from time to time call
         asyncio.sleep(0)  # allow the window to stay active
```

and then run that together with the `rendercanvas.asyncio.loop`.

Would be good to cover this in an example! 🤔 

> might help to have a mode where the canvas is updated only when an event occurs or when any buffer in the scene changes?

No, that's what the "ondemand" mode is for.

**Comment by almarklein** [comment]
> Every so often, write a frame to a video file

In this case, I imagine you either have an offscreen canvas, so you just use `canvas.draw()`. Or you have a regular canvas/renderer and you grab a screenschot from that every somany frames.

> Running a simulation where you need a for-loop for updates/computations

I will create one or two examples for this.

**Comment by almarklein** [comment]
See  #89

**Comment by almarklein** [comment]
Closing, we have `force_draw()` (testable in the `examples/demo.py` which seems to work in all backends.
